tests/test_sequence_feature_extraction_common.py

```diff
@@ -205,12 +205,137 @@ def _inputs_are_equal(input_1, input_2):
             < 1e-3
         )
 
+    def _check_truncation(self, numpify=False):
+        def _inputs_have_equal_length(input):
+            length = len(input[0])
+            for input_slice in input[1:]:
+                if len(input_slice) != length:
+                    return False
+            return True
+
+        def _inputs_are_equal(input_1, input_2):
+            if len(input_1) != len(input_2):
+                return False
+
+            for input_slice_1, input_slice_2 in zip(input_1, input_2):
+                if not np.allclose(np.asarray(input_slice_1), np.asarray(input_slice_2), atol=1e-3):
+                    return False
+            return True
+
+        feat_extract = self.feature_extraction_class(**self.feat_extract_dict)
+        speech_inputs = self.feat_extract_tester.prepare_inputs_for_common(numpify=numpify)
+        input_name = feat_extract.model_input_names[0]
+
+        processed_features = BatchFeature({input_name: speech_inputs})
+
+        # truncate to smallest
+        input_3 = feat_extract.pad(
+            processed_features, padding="max_length", max_length=len(speech_inputs[0]), truncation=True
+        )[input_name]
+        input_4 = feat_extract.pad(processed_features, padding="max_length", max_length=len(speech_inputs[0]))[
+            input_name
+        ]
```

Comment:
Those are pretty hard to read after formatting so the indexing could be done on a separate line.